lib/service/Service.ts

```diff
@@ -694,13 +694,20 @@ class Service extends EventEmitter {
     cancelled$: Observable<void>,
   ) => {
     if (args.existing) {
+      function setAlias(pool: Pool) {
+        return (order: PeerOrder) => {
+          order.alias = pool.getNodeAlias(order.peerPubKey)!;
+          callback(order);
+        };
+      }
+
       this.orderBook.pairIds.forEach((pair) => {
         const ownOrders = this.orderBook.getOwnOrders(pair);
         const peerOrders = this.orderBook.getPeersOrders(pair);
         ownOrders.buyArray.forEach(order => callback(order));
-        peerOrders.buyArray.forEach(order => callback(order));
+        peerOrders.buyArray.forEach(setAlias(this.pool));
```

Comment:
There already is a method in the Service class (`toServiceOrder`) that sets the alias on the order objects returned by these calls, I think it makes sense to reuse that here.

```
ownOrders.buyArray.forEach(order => callback(this.toServiceOrder(order, false)));
peerOrders.buyArray.forEach(order => callback(this.toServiceOrder(order, true)));
ownOrders.sellArray.forEach(order => callback(this.toServiceOrder(order, false)));
peerOrders.sellArray.forEach(order => callback(this.toServiceOrder(order, true)));
```

Then we don't need to make changes to any of the internal order types nor to existing tests.

I also think we need to make the same change to the `orderAdded$.subscribe` call, as it is now I don't believe it would return aliases for newly added orders that happen after the order subscription has begun. There we should do:

```
next: order => callback(this.toServiceOrder(order, true)),
```

Reply:
Thanks for detailed explanation @sangaman you literally implemented this instead of me.